VENTouchLock/VENTouchLock.m

```diff
@@ -174,71 +180,57 @@ - (void)requestTouchIDWithCompletion:(void (^)(VENTouchLockTouchIDResponse))comp
 
 - (void)lock
 {
-    if (![self isPasscodeSet]) {
+    if (![self isPasscodeSet] || self.lockWindow.isKeyWindow) {
         return;
     }
 
     if (self.splashViewControllerClass != NULL) {
         VENTouchLockSplashViewController *splashViewController = [[self.splashViewControllerClass alloc] init];
         if ([splashViewController isKindOfClass:[VENTouchLockSplashViewController class]]) {
-            UIWindow *mainWindow = [[UIApplication sharedApplication].windows firstObject];
-            UIViewController *rootViewController = [UIViewController ventouchlock_topMostController];
-            UIViewController *displayController;
+            self.lockWindow = [[UIWindow alloc] initWithFrame:self.mainWindow.frame];
```

Comment:
The documentation for `-makeKeyAndVisible` says that the method makes sure the window is displayed above all other windows with the same window level or lower.

> This is a convenience method to make the receiver the main window and displays it in front of other windows at the same window level or lower.

To address your concerns, @zinthemoney, I propose doing a few things to guard against becoming obscured and to alert developers to the possibility that their app may be doing something that collides with how VENTouchLock operates.

1. On the invocation of `-lock`, check if `-[UIApplication windows]` has more than 1 window and `-[UIApplication keyWindow]` is not equal to `-[id<UIApplicationDelegate> window]`. If this is the case, we should set `lockWindow`'s `windowLevel` to be equal to the `windowLevel` of the currently key window.
2. Make our lock window a `UIWindow` subclass that overrides `-setRootViewController:` and raises an assertion if the argument is any type of view controller that we don't expect. This won't crash in release builds but will crash in debug builds.
3. Observe the `UIWindowDidBecomeKeyNotification` and `UIWindowDidBecomeVisibleNotification` notifications. If we're currently locked and we receive these messages, raise an assertion that informs the developer that they've attempted to display a window on top of `VENTouchLock` and that this is undefined behavior.